src/types.js

```diff
@@ -47,13 +47,25 @@ export type InputSelectionType = {
 
 export type Account = Auth;
 
-export type ReactionType = any; /* {
+export type SlimEventReaction = {
   emoji_name: string,
   user: any,
-  name: string,
+};
+
+export type EventReaction = {
```

Comment:
You can use a single type and make `emoji_core` and `reaction_type` optional.

Reply:
The problem I experienced with that is that I get an error like this:
```.js
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ src/reactions/aggregateReactions.js:5:3

Cannot return Array.from(...) because:
 • undefined [1] is incompatible with string [2] in property code of array element.
 • undefined [3] is incompatible with string [4] in property type of array element.

     src/reactions/aggregateReactions.js
      2│ import type { EventReaction, AggregatedReaction } from '../types';
      3│
      4│ export default (reactions: EventReaction[], ownEmail: string): AggregatedReaction[] =>
      5│   Array.from(
      6│     reactions
      7│       .reduce((reactionMap, x) => {
      8│         if (!reactionMap.has(x.emoji_name)) {
      9│           reactionMap.set(x.emoji_name, {
     10│             code: x.emoji_code,
     11│             count: 1,
     12│             name: x.emoji_name,
     13│             type: x.reaction_type,
     14│             selfReacted: false,
     15│           });
     16│         } else {
     17│           const prevReaction = reactionMap.get(x.emoji_name);
     18│           if (prevReaction) {
     19│             reactionMap.set(x.emoji_name, {
     20│               ...prevReaction,
     21│               count: prevReaction.count + 1,
     22│             });
     23│           }
     24│         }
     25│
     26│         if (x.user && x.user.email === ownEmail) {
     27│           reactionMap.set(x.emoji_name, {
     28│             ...reactionMap.get(x.emoji_name),
     29│             selfReacted: true,
     30│           });
     31│         }
     32│
     33│         return reactionMap;
     34│       }, new Map())
     35│       .values(),
     36│   );
     37│

     src/types.js
 [1] 51│   emoji_code?: string,
     52│   emoji_name: string,
 [3] 53│   reaction_type?: string,
       :
 [2] 58│   code: string,
     59│   count: number,
     60│   name: string,
     61│   selfReacted: boolean,
 [4] 62│   type: string,
```

So basically, Flow must have a guarantee that `emoji_core` and `reaction_type` are set, or it'll complain. What do you suggest?

Reply:
That is a good point. There are ways to handle this but your approach looks simplest.